app/models/queue_item.rb

```diff
@@ -5,12 +5,28 @@ class QueueItem < ActiveRecord::Base
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  validates :position, numericality: { only_integer: true }
+
   def rating
-    review = Review.where(user_id: user.id, video_id: video.id).first
     review.rating if review
   end
+
+  def rating=(new_rating)
+    if review
+      review.update_column(:rating, new_rating)
+    else
+      review = Review.new(user: user, video: video, rating: new_rating)
+      review.save(validate: false)
+    end
+  end
   
   def category_name
     video.category.name
   end
+
+  private
+
+  def review
+    @review ||= Review.where(user_id: user.id, video_id: video.id).first
```

Comment:
I know, Kevin uses `where(args).first` in the video, but thats because videos were recorded in Rails 3 era. In Rails 4, fetching a single record is preferred via `.find_by`, which is more effective than `where(args).first`.